homework/hm11_bank_card_operations.py

```diff
@@ -0,0 +1,200 @@
+"""
+HOMEWORK11. Bank card operations.
+
+Task Description:
+-----------------
+HOMEWORK11 involves the following task.
+-Write class for bank card.
+-Class should reflect card lifecycle, card operations etc.
+-Use CVV, PIN, Name, Surname , end date, card_num as initial params.
+-Class should have in addition to common logic some class attributes,
+as minimum one class method and as minimum one staticmethod,
+two or more getters/setters, __str_ magic method.
+-Do not forget about block ""if __name__ == '__main__':""
+and code there to check your class written logic.
+"""
+from datetime import datetime
+import random
+
+TERMINAL_SEPARATOR = '-' * 42
+
+
+class BankCard:
+    bank_name = "StereoBank"
+    currency = "USD"
+
+    def __init__(self, name, surname):
+        self.__balance = 0.00
+        self._is_blocked = False
+        self.name = name
+        self.surname = surname
+        self._card_number = self.generate_card_number()
+        self.__pin = self.generate_pin()
+        self.__cvv = self.generate_cvv()
+        self.__expiration_date = self.calculate_end_date()
+
+    @classmethod
+    def generate_card_number(cls):
+        return random.randint(1000000000000001, 9999999999999999)
+
+    @classmethod
+    def generate_pin(cls):
+        return random.randint(0000, 9999)
+
+    @classmethod
+    def generate_cvv(cls):
+        return random.randint(000, 999)
+
+    @classmethod
+    def calculate_end_date(cls):
+        current_date = datetime.now()
+        end_year = current_date.year + 5
+        end_month = current_date.month
+        return f'{end_month:02d}/{end_year}'
+
+    @classmethod
+    def get_bank_info(cls):
+        return f'Bank: {cls.bank_name}, Currency: {cls.currency}'
+
+    @classmethod
+    def get_currency(cls):
+        return cls.currency
+
+    @classmethod
+    def _check_operation_allowed(cls, balance, amount):
+        if balance - amount < 0:
+            raise (ValueError
+                   ('There are not enough funds to perform this operation.'))
+
+    def open_account(self):
```

Comment:
No need in such operation. You already have balance 0.00 after card created

Reply:
right. deleted.